java/ql/lib/semmle/code/java/frameworks/apache/IOGenerated.qll

```diff
@@ -1,684 +1,828 @@
+        "org.apache.commons.io;FilenameUtils;true;getPrefixLength;(String);;Argument[0];generated:regex-use[f]",
+        "org.apache.commons.io;FilenameUtils;true;normalize;(String);;Argument[0];generated:regex-use[f]",
+        "org.apache.commons.io;FilenameUtils;true;normalize;(String,boolean);;Argument[0];generated:regex-use[f]",
+        "org.apache.commons.io;FilenameUtils;true;normalizeNoEndSeparator;(String);;Argument[0];generated:regex-use[f]",
+        "org.apache.commons.io;FilenameUtils;true;normalizeNoEndSeparator;(String,boolean);;Argument[0];generated:regex-use[f]",
+        "org.apache.commons.io;IOUtils;true;contentEquals;(Reader,Reader);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;contentEquals;(Reader,Reader);;Argument[1];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copy;(InputStream,Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copy;(InputStream,Writer,Charset);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copy;(InputStream,Writer,String);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copy;(Reader,Appendable);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copy;(Reader,Appendable,CharBuffer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copy;(Reader,Appendable,CharBuffer);;Argument[2];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copy;(Reader,OutputStream);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copy;(Reader,OutputStream,Charset);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copy;(Reader,OutputStream,String);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copy;(Reader,Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copy;(URL,File);;Argument[0];generated:open-url",
+        "org.apache.commons.io;IOUtils;true;copy;(URL,File);;Argument[1];generated:create-file",
+        "org.apache.commons.io;IOUtils;true;copy;(URL,OutputStream);;Argument[0];generated:open-url",
+        "org.apache.commons.io;IOUtils;true;copyLarge;(Reader,Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copyLarge;(Reader,Writer,char[]);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copyLarge;(Reader,Writer,char[]);;Argument[2];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copyLarge;(Reader,Writer,long,long);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copyLarge;(Reader,Writer,long,long,char[]);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;copyLarge;(Reader,Writer,long,long,char[]);;Argument[4];generated:write-file",
```

Comment:
write-file against Reader, and against `char[]` which is the written data

Reply:
These were derived due to java.io.Writer handwritten write-file sink. I have removed this sink.